File: factorial.js

function factorial(integer) {
  let fact = 1;
  if (integer < 0 || integer === null || integer === "" || integer === undefined || Number.isInteger(integer) === false) {
    console.log("Please enter a non-negative integer and try again!");
  } else if (integer === 0 || integer === 1) {
    console.log(fact);
    return fact;
  } else {
    for (let i = 1; i <= integer; i++) {
      fact = fact * i;
    }
  } 
  console.log(fact);
  return fact;
}

factorial(0);
factorial(1);
factorial(4);
factorial(5);
factorial(6);
factorial(7);
factorial(8);

File: factorial.py

def factorial():
    num = int(input("Enter a positive integer to calculate the factorial:"))
    a = isinstance(num, int)
    if (a == True and num >= 0):
        fact = 1
        for i in range(1, num + 1):
            fact = fact * i
        print("The factorial of {} is {}.".format(num, fact))
    else:
        print("Please enter a positive integer")

factorial()
